---
title: "Microarray Data Analysis"
author: "Mar Navarro Padilla"
date: "6/6/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r Download GEO dataset}
if (!require(GEOquery)) {
  BiocManager::install("GEOquery")
}
require(GEOquery)
gse <- getGEO("GSE70213")
class(gse)
names(gse)
gse[[1]]
esetFromGEO <- gse[[1]]
```

This data was generated by the study _The effect of Nebulin Deficiency on Skeletal Muscle_, that wanted to establish nebulinâ€™s functional roles in adult muscle. It was performed on Neb cKO mice in which there was nebulin deletion driven by the muscle creatine kinase (MCK) promotor and in normal mice. Microarrays investigate the changes in gene expression when nebulin is deficient.

In the study, 4 groups were used: cKO quadriceps, control quadriceps, cKO soleus and control soleus. Each group with 6 replicates, all 6 weeks old mouse males. Quadriceps appeared to be markedly smaller in the Neb cKO mice than in the control, however, soleus is not significantly smaller in Neb cKO mice.

```{r Data summary}
summary(exprs(esetFromGEO))
dim(exprs(esetFromGEO))
```

Within a sample, values of gene expression are very different. This expression set has 35557 rows (genes) and 24 columns (samples).

```{r Plots 1}
hist(exprs(esetFromGEO))
boxplot(exprs(esetFromGEO))
```

Doing a histogram we can see that the expression set is not normalized. The expression values go from 0 to 10000.
```{r Plots 2}
if(!(require(scatterplot3d))) install.packages("scatterplot3d")
require(scatterplot3d)
x <- exprs(esetFromGEO) 
pcX<-prcomp(t(x), scale=FALSE) # o prcomp(t(X))
res3d<-scatterplot3d(pcX$x[,1:3], angle=20)
manDist <- dist(t(x))
heatmap (as.matrix(manDist), col=heat.colors(16))
```

From the 3D scatterplot and the heatmap we can observe that samples are not correlated.

# Experimental design
* Response variable: gene expression
* Groups and levels: 2 factors (nebulin and muscle), each at 2 levels (present or deleted, quadriceps or soleus)
* Sample size and number of experimental units per combination of factors: 24 samples and 6 replicates
* Type of replicates: biological
* Type of design: block design

As the question in the study is _Which is the effect of nebulin in muscle?_, a comparison between the group with nebulin in quadriceps and the group where nebulin in quadriceps is deleted has to be performed, as well as a comparison between samples of soleus of Neb CKO mice and samples of normal mouse soleus.

```{r Contrast matrix}
a <- c(1,0)
b <- c(-1,0)
c <- c(0,1)
d <- c(0,-1)
table <- data.frame(a,b,c,d)
colnames(table) <- c("alpha 1 (nebulin quadriceps)","alpha 2 (no nebulin quadriceps)","alpha 3 (nebulin soleus)","alpha 4 (no nebulin soleus")
rownames(table) <- c("alpha 1 - alpha 2", "alpha 3 - alpha 4")
table
```

```{r Design matrix and linear model}
require(affy)
require(Biobase)
targets <- pData(esetFromGEO)
treat<- targets$description
lev<-factor(treat, levels=unique(treat))
design <-model.matrix(~0+lev)
colnames(design)<-c('normal.quadriceps','nebulin.deficient.quadriceps','normal.soleus','nebulin.deficient.soleus')
print(design)

library(limma)
cont.matrix1 <- makeContrasts(normal.quadriceps-nebulin.deficient.quadriceps,
        levels=design)
fit1<-lmFit(esetFromGEO, design)
fit.main1<-contrasts.fit(fit1, cont.matrix1)
fit.main1<-eBayes(fit.main1)
fit.main1

topTab <- topTable(fit.main1, number=nrow(fit.main1), adjust="fdr",lfc=abs(3))
topTab

volcanoplot(fit.main1, highlight=10, names=fit.main1$ID, 
            main=paste("Differentially expressed genes",colnames(cont.matrix1), sep="\n"))
```

```{r Second comparison}
cont.matrix2 <- makeContrasts(normal.soleus-nebulin.deficient.soleus,
        levels=design)
fit2<-lmFit(esetFromGEO, design)
fit.main2<-contrasts.fit(fit1, cont.matrix2)
fit.main2<-eBayes(fit.main2)
fit.main2

topTab2 <- topTable(fit.main2, number=nrow(fit.main2), adjust="fdr",lfc=abs(3))
topTab

volcanoplot(fit.main2, highlight=10, names=fit.main2$ID, 
            main=paste("Differentially expressed genes",colnames(cont.matrix2), sep="\n"))
```

The gene identifiers in the top table correspond to the differentially expressed genes. The blue gene identifiers in the volcano plot are the differentially expressed genes that are more significant. As there are differentially expressed genes we can say that there are differences in the muscles when nebulin is deficient. Moreover, it seems that there are more significantly differentially expressed genes in the first comparison, that is, in the quadriceps.